src/lager.erl

```diff
@@ -594,3 +595,34 @@ pr_stacktrace(Stacktrace) ->
 pr_stacktrace(Stacktrace, {Class, Reason}) ->
     lists:flatten(
         pr_stacktrace(Stacktrace) ++ "\n" ++ io_lib:format("~s:~p", [Class, Reason])).
+
+rotate_file(FileName) ->
+    Handlers = lager_config:global_get(handlers),
+    RotateHandlers = lists:filter(
+        fun ({{lager_file_backend, FN}, _, _}) ->
```

Comment:
Worth considering whether this should be an underscore match against the module name so that other file-wrapping handlers could be used. If so, `lager_util:expand_path` would need to gracefully handle non-string arguments, or those would have to be wrapped in catch.

Alternatively or additionally, exposing `rotate_handler` (perhaps with a parameter list that makes it look more like `rotate_file`) would allow other backends to participate in this (without the user having to invoke `rotate_sink` or `rotate_all` to sidestep our constraints).